internal/pkg/servicebus/amqpconnection.go

```diff
@@ -180,10 +184,85 @@ func NewAmqpConnection(ctx context.Context, config *types.Configuration) *AmqpCo
 
 	listener.Session = createAmqpSession(&listener)
 	listener.Receiver = createAmqpListener(&listener)
+	listener.ManagementSender, listener.ManagementReceiver, err = listener.createAmqpSBManagementChannels(listener.TopicName, config.ModuleName)
+	if err != nil {
+		log.WithError(err).Error("failed to create management sender, without this renewal of message locks will fail")
+	}
 
 	return &listener
 }
 
+//RenewLocks renews the locks on messages provided
+func (l *AmqpConnection) RenewLocks(ctx context.Context, messages []*amqp.Message) error {
+	lockTokens := make([]amqp.UUID, 0, len(messages))
+	for _, m := range messages {
+		lockTokens = append(lockTokens, m.DeliveryAnnotations["x-opt-lock-token"].(amqp.UUID))
```

Comment:
Should we error check for existence of "x-opt-lock-token" and then check the type assertion is successful in order to raise appropriate errors or just it panic?

Reply:
Done